src/ompl/geometric/planners/bitstar/src/BITstar.cpp

```diff
@@ -0,0 +1,1913 @@
+            }
+
+            if (hasSolution_ == true)
+            {
+                this->endSuccessMessage();
+
+                this->publishSolution();
+            }
+            else
+            {
+                this->endFailureMessage();
+            }
+
+            this->statusMessage(ompl::msg::LOG_DEBUG, "End solve");
+
+            //PlannerStatus(addedSolution, approximate)
+            return ompl::base::PlannerStatus(hasSolution_, approximateSoln_);
+        }
+
+
+
+        void BITstar::getPlannerData(ompl::base::PlannerData& data) const
+        {
+            //Get the base planner class data:
+            Planner::getPlannerData(data);
+
+            //Add samples
+            if (freeStateNN_)
+            {
+                //Variables:
+                //The list of unused samples:
+                std::vector<VertexPtr> samples;
+
+                //Get the list of samples
+                freeStateNN_->list(samples);
+
+                //Iterate through it turning each into a disconnected vertex
+                for (std::vector<VertexPtr>::const_iterator sIter = samples.begin(); sIter != samples.end(); ++sIter)
+                {
+                    //No, add as a regular vertex:
+                    data.addVertex(ompl::base::PlannerDataVertex((*sIter)->stateConst()));
+                }
+            }
+            //No else.
+
+            //Add vertices
+            if (vertexNN_)
+            {
+                //Variables:
+                //The list of vertices in the graph:
+                std::vector<VertexPtr> vertices;
+
+                //Get the list of vertices
+                vertexNN_->list(vertices);
+
+                //Iterate through it turning each into a vertex with an edge:
+                for (std::vector<VertexPtr>::const_iterator vIter = vertices.begin(); vIter != vertices.end(); ++vIter)
+                {
+                    //Is the vertex the start?
+                    if ((*vIter)->isRoot() == true)
+                    {
+                        //Yes, add as a start vertex:
+                        data.addStartVertex(ompl::base::PlannerDataVertex((*vIter)->stateConst()));
+                    }
+                    else
+                    {
+                        //No, add as a regular vertex:
+                        data.addVertex(ompl::base::PlannerDataVertex((*vIter)->stateConst()));
+
+                        //And as an incoming edge
+                        data.addEdge(ompl::base::PlannerDataVertex((*vIter)->getParentConst()->stateConst()), ompl::base::PlannerDataVertex((*vIter)->stateConst()));
+                    }
+                }
+            }
+            //No else.
+
+            //Did we find a solution?
+            if (hasSolution_ == true)
+            {
+                data.markGoalState(goalVertex_->stateConst());
+            }
+
+            data.properties["best_solution_cost DOUBLE"] = this->bestCostProgressProperty();
```

Comment:
Oh, is this not necessary? I assumed I'd copied this from `RRT*` as an example, but that does not appear to be the case (at least anymore), so I'm not sure where I got this idea from... I did think it was a little weird that I had to do it twice. :wink: